src/legacy/core_plugins/interpreter/common/types/kibana_datatable.ts

```diff
@@ -19,30 +19,49 @@
 
 import { map } from 'lodash';
 
+const name = 'kibana_datatable';
+
+interface Column {
+  id: string;
+  name: string;
```

Comment:
Does the column not contain any type information? That'd be pretty handy to have in a number of situations. (e.g. `type: 'string'`)

Reply:
I don't think so -- kibana_datatable is just the tabified data format, e.g.

```javascript
{
  columns: [{ id: 'foo', name: 'Foo Col' }],
  rows: [{ foo: 'bar' }, { foo: 'baz' }],
}
```

Reply:
Chris, yes it'd be handy, it's not present at the moment, but typed table attribs had come up in discussion. For example, in the context of TS, it could warn that one wants to multiply a numeric column with an alpha column, or filter a time attribute represented as milliseconds since epoch with a range that represents time as days since epoch (ie. the physical type matches but the _attribute domains_ have mismatched semantics).

Also, semantic types would be great for something like.... helping determine Kibana Lens options :-) For example, if an editor knew that those numbers represent time in milliseconds, then appropriate time filters etc. can be added (example is made up, only to illustrate that physical, easy to infer types such as numbers don't help a downstream, uncoupled consumer very much). On the demo day prez for shareable dataviz I touched on using attribute types as a goal, and super early steps.

Reply:
@chrisdavies ^ I should've used the handle